libs/cases/frontend/view/src/lib/components/detail-case/detail-case.component.ts

```diff
@@ -27,5 +30,135 @@ export class DetailCaseComponent {
         return _case.find((_case) => String(_case.id) === this.route.snapshot.params['id']);
     });
 
-    constructor(private route: ActivatedRoute) {}
+    isEditing = false;
+    wasValidated = false;
+    datePattern = /^\d{4}-\d{2}-\d{2}T00:00:00\.000Z$/;
+
+
+    constructor(private route: ActivatedRoute, private readonly _modalService: ModalService, private readonly toastService: ToastService) {
+    }
+
+    deleteCase() {
+        const caseId = this.mapCaseId(this.casedetail());
+        if (caseId !== undefined) {
+            this._browseFacade.deleteCase(caseId).subscribe();
+        }
+    }
+
+    mapCaseId(_case: ICaseResponse | undefined) {
+        if (_case === undefined) {
+            return undefined;
+        }
+        return {
+            id: _case.id,
+        };
+    }
+
+    async deleting() {
+        const instance = await this._modalService.open({
+            content: DeleteModalComponent,
+        });
+
+        // modal closes on confirm and dismisses on cancel
+        instance.onClose.on(() => {
+            this.deleteCase();
+        });
+    }
+
+    enableEditing() {
+        this.isEditing = true;
+    }
+
+    cancelEdit() {
+        this.isEditing = false;
+    }
+
+    toggleEdit() {
+        if (this.isEditing) {
+            this.onSubmit();
+        } else {
+            this.isEditing = true;
+        }
+    }
+
+    onSubmit(): void {
+        // TODO validate form
+        if (this.validateForm() === 'valid') {
+            const caseId = this.mapCaseId(this.casedetail());
+            const caseData = this.casedetail();
+
+            if (caseId !== undefined && caseData !== undefined) {
+                this._browseFacade.updateCase(caseId, caseData).subscribe({});
+            }
+            this.isEditing = false;
+        } else {
+            this.showErrorToast(this.validateForm());
```

Comment:
validateForm is called twice, once in line 86 and once here. Save the result in a const instead

Reply:
Actually it would be good to get the casedetail at the start of the "onSubmit" function and pass it as an argument.

const casedetail = this.casedetail();
const validationMessage = this.validateForm(casedetail);
...

Reply:
I fixed the first one and think that the second one is not super necessary.

Reply:
I think it would be good. Right now you might validate a different casedetail than the one you submit. Just move the const caseData to the start of the function and then pass it to the validateForm and also use it for the caseId

Reply:
Done